src/main/java/org/threadly/concurrent/SingleThreadScheduler.java

```diff
@@ -282,20 +294,28 @@ protected void shutdownService() {
      * @return if {@code stopImmediately} is {@code true}, this will include tasks which were queued to run, 
      *         otherwise will be an empty list
      */
-    protected List<Runnable> stop(boolean stopImmediately) {
-      if (stopIfRunning()) {
-        if (stopImmediately) {
-          return finishShutdown();
+    public List<Runnable> stop(boolean stopImmediately) {
```

Comment:
This stop function is actually better than it ever has been....stopIfRunning() would have been false if it was never started, thus making it so we never transition to shutdown finished previously.